executor/table_reader.go

```diff
@@ -72,9 +72,10 @@ type TableReaderExecutor struct {
 	ranges []*ranger.Range
 
 	// kvRanges are only use for union scan.
-	kvRanges []kv.KeyRange
-	dagPB    *tipb.DAGRequest
-	startTS  uint64
+	kvRanges       []kv.KeyRange
+	kvRangesBackup []kv.KeyRange
```

Comment:
A better way is moving the `buildResp` out of `Open` to `Next`.

Reply:
A `Backup` makes me feel weird.

Reply:
I am feeling ackward as well, perhaps a better name like `initialKvRanges` would make it less weird, while I did not quite get it what you mean by moving the buildResp to `Next`, and I don't think it would solve the problem here. The problem is, whenever `buildResp` is called, new ranges will be added to `kvRanges`, even if it is put under `Next`, as long as it is not cleaned, the redundant ranges will be remained.